public/app/features/dashboard/components/ShareModal/SharePublicDashboard.tsx

```diff
@@ -38,32 +51,142 @@ export const SharePublicDashboard = (props: Props) => {
+      setAcknowledgements({ ...acknowledgements, [field]: checked });
+    },
+    [acknowledgements]
+  );
+
+  // check if all conditions have been acknowledged
+  const acknowledged = () => {
+    return (
+      acknowledgements.public && acknowledgements.datasources && acknowledgements.variables && acknowledgements.usage
+    );
   };
 
   return (
     <>
+      <p>Welcome to Grafana public dashboards alpha!</p>
+      <p>
+        To allow the current dashboard to be published publicly, toggle the switch. For now we do not support template
+        variables or frontend datasources.
+      </p>
       {dashboardHasTemplateVariables(dashboardVariables) && (
         <Alert severity="warning" title="dashboard cannot be public">
           This dashboard cannot be made public because it has template variables
         </Alert>
       )}
-
-      <p className="share-modal-info-text">Public Dashboard Configuration</p>
-
-      <Field label="Enabled" description="Configures whether current dashboard can be available publicly">
-        <Switch
-          id="share-current-time-range"
-          disabled={dashboardHasTemplateVariables(dashboardVariables)}
-          value={publicDashboardConfig?.isPublic}
-          onChange={() =>
-            setPublicDashboardConfig((state) => {
-              return { ...state, isPublic: !state.isPublic };
-            })
-          }
-        />
-      </Field>
-      <Button onClick={onSavePublicConfig}>Save Sharing Configuration</Button>
+      We&apos;d love your feedback. To share, please comment on this{' '}
+      <a
+        href="https://github.com/grafana/grafana/discussions/49253"
+        target="_blank"
+        rel="noreferrer"
+        className="text-link"
+      >
+        github discussion
+      </a>
+      <hr />
+      {!publicDashboardPersisted(publicDashboard) && (
+        <div>
+          Before you click Save, please acknowledge the following information: <br />
+          <FieldSet>
+            <br />
+            <div>
+              <Checkbox
+                label="Your entire dashboard will be public"
+                value={acknowledgements.public}
+                onChange={(e) => onAcknowledge('public', e.currentTarget.checked)}
+              />
+            </div>
+            <br />
+            <div>
+              <Checkbox
+                label="Publishing currently only works with a subset of datasources"
+                value={acknowledgements.datasources}
+                onChange={(e) => onAcknowledge('datasources', e.currentTarget.checked)}
+              />
+              <Icon name="info-circle" />
+            </div>
+            <br />
+            <Checkbox
+              label="Variables can be sensitive and are currently not recommended"
+              value={acknowledgements.variables}
+              description="learn more about template variables"
+              onChange={(e) => onAcknowledge('variables', e.currentTarget.checked)}
+            />
```

Comment:
this text is strange. In what way is it sensitive? Any use of variables in queries will simply not work. Should we even allow making a dashboard public that has variables?

Reply:
>Should we even allow making a dashboard public that has variables?

We have a story of doing this in the next milestone. I agree the copy could use some edits.

Reply:
Fixed